src/main/java/org/mozilla/taskcluster/client/temporarycredentials/TemporaryCredentials.java

```diff
@@ -0,0 +1,108 @@
+package org.mozilla.taskcluster.client;
+
+import java.util.Date;
+import java.security.NoSuchAlgorithmException;
+import java.security.InvalidKeyException;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+
+import net.iharder.Base64;
+
+import org.mozilla.taskcluster.client.InvalidOptionsException;
+
+/*
+  Temporary Credentials for taskcluster
+  For creating unnamed Temporary credentials :
+    @params
+    clientId :String
+    accessToken :String
+    scopes :String[]
+    start :Date
+    expiry :Date
+*/
+
+public class TemporaryCredentials{
```

Comment:
Here `TemporaryCredentials` exposes static methods for returning temporary credentials as strings. Would it instead be possible for these to be constructors or factory methods inside the `Credentials` class?